src/store/index.js

```diff
@@ -61,6 +96,58 @@ export default createStore({
       });
       commit(SET_MESSAGES, response.content);
     },
+    async fetchAllMessages({ commit }) {
+      const response = await MessagesService.fetchAllMessages();
+      commit(SET_ALL_MESSAGES, response);
+    },
+
+    async fetchSchedules({ commit }, pageNumber) {
+      const response = await SchedulesService.fetchSchedules(pageNumber);
+      const schedules = response.content;
+      schedules.forEach((schedule) => {
+        schedule.nextRun = FormatUtils.formatDate(
+          schedule.nextRun,
+          'dd MMM yyyy'
+        );
+        schedule.activeLabel = schedule.active ? 'Active' : 'Inactive';
+      });
+
+      // set pagination
+      commit(SET_PAGINATION, {
+        page: response.pageable.pageNumber + 1,
+        total: response.totalElements,
+      });
+      commit(SET_SCHEDULES, schedules);
+    },
+    async deleteSchedule({ commit }, id) {
+      await SchedulesService.deleteSchedule(id);
+
+      // set pagination
+      commit(DECREMENT_PAGINATION_TOTAL);
+      commit(REMOVE_SCHEDULE, id);
+    },
+    async createSchedule({ commit }, schedule) {
+      await SchedulesService.createSchedule(schedule);
+      commit(CLOSE_APP_MODAL);
+    },
+    fetchSchedule({ state }, id) {
+      const schedule = state.schedules.filter((schedule) => {
+        return schedule.scheduleId === id;
+      });
+      return schedule[0];
+    },
+    async editSchedule({ commit }, { id, schedule }) {
```

Comment:
You already have the id inside the schedule object

also update related logic
```suggestion
    async editSchedule({ commit }, schedule) {
```